src/util/fs.js

```diff
@@ -163,6 +163,7 @@ async function buildActionsForCopy(
     const {src, dest, type} = data;
     const onFresh = data.onFresh || noop;
     const onDone = data.onDone || noop;
+    invariant(!files.has(dest), `The same file ${dest} can't be copied twice in one bulk copy`);
```

Comment:
Why would it be?
Set should be implemented by a HashSet and its has operation should be sublinear, O(1) in an ideal situation.
https://jsperf.com/array-indexof-vs-set-has